python/az/aro/azext_aro/_actions.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the Apache License 2.0.
+
+import argparse
+
+from azext_aro.vendored_sdks.azure.mgmt.redhatopenshift.v2024_08_12_preview.models import PlatformWorkloadIdentity
+from azure.cli.core.azclierror import CLIError
+
+
+# pylint:disable=protected-access
+# pylint:disable=too-few-public-methods
+class AROPlatformWorkloadIdentityAddAction(argparse._AppendAction):
```

Comment:
Was using validator not enough?
It looks to me that we can achieve the same with a validator and some formalizations after the parse.

I think we shouldn't do `protected-access` because it might be broken some day.

Reply:
This implementation follows the Azure CLI recommendations for implementing "action"-based parameters like `--assign-platform-workload-identity`: https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#collection-management-with-actions 

It might be possible to do this with a custom implementation, but given that this code will eventually be upstreamed to the Azure CLI I'd prefer to just follow their conventions.

Reply:
I didn't know there's a doc about it. thanks!